### ISSUE TITLE
AudioSegment.export doesn't always close file

### ISSUE THREAD

**Issue by p-i-** [jiaaro/pydub#305]
I'm getting `ResourceWarning: unclosed file`

https://github.com/jiaaro/pydub/blob/master/pydub/audio_segment.py#L737
^ Not all execution paths close `out_f`

Maybe consider using:

`with open(somefile) as f:`

**Comment by craigthelinguist** [comment]
I think that's by design? if you `audio_seg.export("out.wav")`, it creates and opens a file "out.wav", writes to it, then gives you the open file handle. I agree it's surprising. It would be nice to make that behaviour clear in the docstring/supporting material. 

I think it also means that if `export` throws an error and you catch and recover from it, `out_f` will still be open, but you won't have a reference to it, potentially causing a leak.

**Comment by p-i-** [comment]
I don't see why you would want the open file handle tho'. What would you do with it? 

Also it's important that you're guaranteed not to have a dangling file handle. A `save` operation should encapsulate the file handle. That is an implementation detail.

**Comment by jiaaro** [comment]
I agree it was an unfortunate design decision that I made in v0.0.1 and is still with us. Maybe in v1.0 we can change this

**Comment by partounian** [comment]
Any update on this?

**Comment by opencode-source** [comment]
you can easily solve this problem

out_ = AudioSegment.export("filename.wav", format="wav")
out_.close()

**Comment by ProjectDarkTiger** [comment]
> you can easily solve this problem
> 
> out_ = AudioSegment.export("filename.wav", format="wav")
> out_.close()

That's awesome! It fixed my problem. Thank you!

**Comment by mattjcamil** [comment]
adding .close() return the following error

AttributeError: 'AudioSegment' object has no attribute 'close'

**Comment by kidney-bean** [comment]
> you can easily solve this problem
> 
> out_ = AudioSegment.export("filename.wav", format="wav")
> out_.close()

this doesnt work...
